### DISCUSSION TITLE
Does save file have characters ID?

### DISCUSSION THREAD

**Discussion by johndoe71rus** [diasurgical/DevilutionX#1273] [Q&A] [answered]
maybe this is not a issue.
1 start the game.
2 save the character, save the file single_0.hsv.
3 gain the level of the character.
4 save new character to file single_0.hsv then rename it to single_1.hsv
5 copy single_1.hsv next to old single_0.hsv
6 start the game, see 2 characters with different levels
7 load character from single_1.hsv, he has a higher level.

As a result, it will load the player from single_0.hsv
Does some character ID store inside save file?

**Comment by qndel** [comment, marked as answer]
I think it's about the characters having the same name that's causing the issue

**Comment by AJenbo** [comment]
the game will search for a save withe the given hero name. 0 comes before 1.

**Comment by johndoe71rus** [comment]
Before loading, the game distinguishes between different saves with the same name.
![cap_vokoscreenNG-2021-03-25_10-13-00_00:00:04_02](https://user-images.githubusercontent.com/7861172/112510725-c2f53400-8da2-11eb-9dc7-0c74433d2bf7.jpg)
![cap_vokoscreenNG-2021-03-25_10-13-00_00:00:02_01](https://user-images.githubusercontent.com/7861172/112510740-c7215180-8da2-11eb-82da-a52e912526b4.jpg)

But after selecting and loading the game, it only loads save_0.
Is it a bad idea to store savefiles the hero at different levels? Or is it possible to think about enhancent.

**Reply by AJenbo** [reply]
Don't have multiple heroes with the same name, it's not supported, ever was might ever be.

The UI doesn't allow you to do this, so it can only happen if you manually mess with the files.
